cwms-data-api/src/main/java/cwms/cda/api/watersupply/WaterPumpDeleteController.java

```diff
@@ -0,0 +1,147 @@
+    private static final Logger LOGGER = Logger.getLogger(WaterPumpDeleteController.class.getName());
+    private static final String PUMP_TYPE = "pump-type";
+
+    public WaterPumpDeleteController(MetricRegistry metrics) {
+        waterMetrics(metrics);
+    }
+
+    @OpenApi(
+        queryParams = {
+            @OpenApiParam(name = PUMP_TYPE, required = true, type = PumpType.class,
+                    description = "The type of pump to be removed from the contract."
+                            + " Expected values: IN, OUT, OUT BELOW"),
+            @OpenApiParam(name = DELETE, type = boolean.class, required = true,
+                    description = "Whether to delete the associated accounting data.")
+        },
+        pathParams = {
+            @OpenApiParam(name = NAME, description = "The name of the pump to be "
+                    + "removed from the specified contract.", required = true),
+            @OpenApiParam(name = OFFICE, description = "The office the project is associated with.", required = true),
+            @OpenApiParam(name = PROJECT_ID, description = "The name of the project.", required = true),
+            @OpenApiParam(name = CONTRACT_NAME, description = "The name of the contract the pump is associated with.",
+                    required = true),
+            @OpenApiParam(name = WATER_USER, description = "The name of the water user the contract "
+                    + "is associated with.", required = true)
+        },
+        responses = {
+            @OpenApiResponse(status = "404", description = "The provided combination of parameters"
+                    + " did not find any contracts."),
+            @OpenApiResponse(status = "501", description = "Requested format is not implemented.")
+        },
+        description = "Delete a pump from a contract",
+        path = "/projects/{office}/{project-id}/water-user/{water-user}/contracts/{contract-id}/pumps/{name}",
+        method = HttpMethod.DELETE,
+        tags = {TAG}
+    )
+
+    @Override
+    public void handle(@NotNull Context ctx) {
+        try (Timer.Context ignored = markAndTime(DELETE)) {
+            DSLContext dsl = getDslContext(ctx);
+            boolean deleteAccounting = Boolean.parseBoolean(ctx.queryParam(DELETE));
+            String officeId = ctx.pathParam(OFFICE);
+            String projectName = ctx.pathParam(PROJECT_ID);
+            String entityName = ctx.pathParam(WATER_USER);
+            PumpType pumpType = PumpType.valueOf(ctx.queryParam(PUMP_TYPE));
+            String contractName = ctx.pathParam(CONTRACT_NAME);
+            WaterContractDao contractDao = getContractDao(dsl);
+            CwmsId projectLocation = buildCwmsId(officeId, projectName);
+            List<WaterUserContract> contract = contractDao.getAllWaterContracts(projectLocation, entityName);
+
+            for (WaterUserContract waterUserContract : contract) {
```

Comment:
This delete is not meant to delete from multiple contracts, it should be a single retrieve instead.

Reply:
Replaced with recently added single retrieve function